### PR TITLE
feature/BHT-9-wallet-connect

### PR DESCRIPTION
**Pull request by abbas-bictory**

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -6,6 +6,7 @@ node_modules
 .env
 .env.*
 !.env.example
+static/flexbox.min.css
```
Comment (amirbictory) [inline .prettierignore:9]: please remove this dependency ! we are using tailwindcss as css framework

File: src/lib/long-text/long-text.svelte [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<script lang="ts">
```
Comment (amirbictory) [inline src/lib/long-text/long-text.svelte:1]: We have a components folder inside lib folder, please put components inside lib/components and if they are single files don't need to add a folder , just use the component name.

File: src/lib/long-text/long-text.svelte [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  let slotContent: string;
+  export let visibleLength: number | [number, number] = 5;
```
Comment (amirbictory) [inline src/lib/long-text/long-text.svelte:3]: why two types ?

File: src/lib/wallet-connect/wallet-connect.svelte [hunk 4]
```diff
@@ -0,0 +1,213 @@
+<script lang="ts">
+  import { PUBLIC_WALLET_CLUSTER_API_URL } from '$env/static/public';
+  import Icon from '@iconify/svelte';
```
Comment (amirbictory) [inline src/lib/wallet-connect/wallet-connect.svelte:3]: please sync with team about any added library , i think this one is fine .

File: src/lib/wallet-connect/wallet-connect.svelte [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  import { walletStore } from '@svelte-on-solana/wallet-adapter-core';
+
+  export let walletOwnerAvatar: string = '';
+
+  const localStorageKey = 'walletAdapter';
+  const network = clusterApiUrl(PUBLIC_WALLET_CLUSTER_API_URL as Cluster);
+  let wallets: any[];
+
+  onMount(async () => {
+    const { PhantomWalletAdapter, SolflareWalletAdapter, TorusWalletAdapter } = await import(
```
Comment (amirbictory) [inline src/lib/wallet-connect/wallet-connect.svelte:23]: why doing this inside onMount ? you are initializing wallets in the end .

File: src/lib/wallet-connect/wallet-connect.svelte [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+        node &&
+        event.target instanceof Node &&
+        !node.contains(event.target) &&
+        !event.defaultPrevented
+      ) {
+        callbackFunction();
+      }
+    }
+
+    document.body.addEventListener('click', onClick, true);
```
Comment (amirbictory) [inline src/lib/wallet-connect/wallet-connect.svelte:102]: svelte has a component for body to add event.

<svelte:body on:click={clickEvent} />

**Review by amirbictory — COMMENTED** [review]
Whole styling of components should be refactored since we are using tailwindcss in the oproject.

**Review by amirbictory — CHANGES_REQUESTED** [review]
please refactor styling completely and use tailwindcss.

File: src/lib/long-text/long-text.svelte (hunk 3 above)
Comment (abbas-bictory) [reply src/lib/long-text/long-text.svelte:3]: This components will take a long string (like wallet address) and turns it into 0xAbcd....xyz (string+dots+string)
the visibleLength controls number of visible characters before and after dots. If you just pass a number, then the number of chars before and after dots are similar, however you can have more control over it by an array. So visibleLength =[7, 3], will show 7 chars at beginning, then some dots, and finally 3 chars at the end

File: yarn.lock
Comment (amirbictory) [inline yarn.lock:1]: please remove this file and only use pnpm ,

**Review by amirbictory — CHANGES_REQUESTED** [review]

**Pull request merged**